resolver/custom_dns_resolver.go

```diff
@@ -120,26 +137,45 @@ func (r *CustomDNSResolver) processRequest(request *model.Request) *model.Respon
 		}
 	}
 
-	return nil
+	logger.WithField("next_resolver", Name(r.next)).Trace("go to next resolver")
+	forwardResponse, _ := r.next.Resolve(ctx, request)
+
+	return forwardResponse
 }
 
 // Resolve uses internal mapping to resolve the query
 func (r *CustomDNSResolver) Resolve(ctx context.Context, request *model.Request) (*model.Response, error) {
-	logger := log.WithPrefix(request.Log, "custom_dns_resolver")
-
 	reverseResp := r.handleReverseDNS(request)
 	if reverseResp != nil {
 		return reverseResp, nil
 	}
 
-	if len(r.mapping) > 0 {
-		resp := r.processRequest(request)
-		if resp != nil {
-			return resp, nil
-		}
+	resp := r.processRequest(ctx, request)
+	return resp, nil
+}
+
+func (r *CustomDNSResolver) processIP(ip net.IP, question dns.Question) (result []dns.RR) {
+	result = make([]dns.RR, 0)
+
+	if isSupportedType(ip, question) {
+		rr, _ := util.CreateAnswerFromQuestion(question, ip, r.cfg.CustomTTL.SecondsU32())
```

Comment:
I don't think this error will ever happen, but since we're changing the other functions to return an error, can you  check and return it if it occurs?